client/tx/tx.go

```diff
@@ -257,24 +258,28 @@ func BuildSimTx(txf Factory, msgs ...sdk.Msg) ([]byte, error) {
 	// sentinel pubkey.
 	sig := signing.SignatureV2{}
 
-	if err := tx.SetSignatures(sig); err != nil {
+	Sign(txf, from, txb)
```

Comment:
this works, but i think it's slightly overkill, as in simulate mode we never verify sigs.

Does it work if you remove this line and use
```go
sig := signing.SignatureV2{
		Data: &signing.SingleSignatureData{
			SignMode: txf.signMode,
		},
		Sequence: txf.Sequence(),
	}
```

I think you won't even need the `from` arg in this function, nor the SetSignatures below.

Reply:
perf. thank you for a more elegant solution here. Will give it a go